extension/src/popup/ducks/transactionData.ts

```diff
@@ -0,0 +1,36 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+interface TransactionData {
+  amount: string;
+  asset: string;
+  destination: string;
+  transactionFee: string;
+  memo: string;
+}
+
+const initialState: TransactionData = {
+  amount: "",
+  asset: "native",
+  destination: "",
+  // TODO - use lumens instead of stroops
+  transactionFee: "100",
+  memo: "",
+};
+
+const transactionDataSlice = createSlice({
+  name: "transactionData",
+  initialState,
+  reducers: {
+    saveDestination: (state, action) => {
+      state.destination = action.payload;
+    },
+    // TODO - add for each field
+  },
+});
+
+export const { saveDestination } = transactionDataSlice.actions;
+export const { reducer } = transactionDataSlice;
+
+export const transactionDataSelector = (state: {
+  transactionData: TransactionData;
+}) => state.transactionData;
```

Comment:
Hm, what do you think of coupling this with the `transactionSubmissionSlice`? They're tied together as we should only have one transaction in process at a time (ideally). 

for ex:
```
initalState = {
status,
data: {amount, asset, etc. },
xdr,
}
```

Once the the transaction has terminated, you could clear the whole slice to reset.

Reply:
I like that idea a lot, moved here:
https://github.com/stellar/freighter/pull/396/commits/bb0ff06a4a5af591fd1bbe335818743b4be73b48